### PR TITLE
feat(query): add support for limit function in queries and label names API

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

- Support for `limit` function, e.g `expression limit <num-rows>`
- Support for label-names API [Prometheus label names](https://prometheus.io/docs/prometheus/latest/querying/api/#getting-label-names) - `/api/v1/labels?filter=<column-filters>`

### CODE REVIEW

File: core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala [hunk 1]
```diff
@@ -522,6 +527,15 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     collector.result
   }
 
+  def labelNamesFromFilters(columnFilters: Seq[ColumnFilter],
+                          startTime: Long,
+                          endTime: Long,
+                          limit: Int): Seq[Int] = {
+    val partIdsOrderedByEndTime = new TopKPartIdsCollector(limit)
+    searchFromFilters(columnFilters, startTime, endTime, partIdsOrderedByEndTime)
+    partIdsOrderedByEndTime.topKPartIDsBitmap().toArray
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala:536]: Top-k is an expensive query. Can we do a simple search and get 1 result ?

File: query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala [hunk 2]
```diff
@@ -65,6 +65,11 @@ final case class PartKeysDistConcatExec(queryContext: QueryContext,
 final case class LabelValuesDistConcatExec(queryContext: QueryContext,
                                            dispatcher: PlanDispatcher,
                                            children: Seq[ExecPlan]) extends MetadataDistConcatExec
+final case class LabelNamesDistConcatExec(queryContext: QueryContext,
+                                           dispatcher: PlanDispatcher,
+                                           children: Seq[ExecPlan]) extends DistConcatExec {
+  addRangeVectorTransformer(LimitFunctionMapper(1))
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala:71]: Let us not add transformers during runtime. Can implement this in do/execute methods.

**Review by vishramachandran — APPROVED** [review]
Approved with important comments.

File: core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala (hunk 1 above)
Comment (sherali42) [reply core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala]: Done 👍 , updated code to use SinglePartIdCollector

**Pull request merged**